File: UsuariosAPI/src/main/java/com/clinica/api/personal_service/repository/DoctorRepository.java
package com.clinica.api.personal_service.repository;

import com.clinica.api.personal_service.model.Doctor;
import java.util.List;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface DoctorRepository extends JpaRepository<Doctor, Long> {

    List<Doctor> findByActivoTrue();

    Optional<Doctor> findByIdAndActivoTrue(Long id);

    Optional<Doctor> findByUsuario_IdAndActivoTrue(Long usuarioId);
}

File: usuario-service/src/main/kotlin/com/clinica/api/usuario_service/repository/DoctorRepository.kt
package com.clinica.api.usuario_service.repository

import com.clinica.api.usuario_service.model.Doctor
import com.clinica.api.usuario_service.model.Usuario
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface DoctorRepository : JpaRepository<Doctor, Long> {
    fun findByUsuario(usuario: Usuario): Optional<Doctor>
    fun deleteByUsuario(usuario: Usuario)
    fun existsByUsuario(usuario: Usuario): Boolean
}
